blog/_posts/2021-11-15-k8s-custom-deploy.md

```diff
@@ -0,0 +1,106 @@
+---
+slug: "k8s-custom-deploy"
+date: 2021-11-18
+author: milas
+layout: blog
+title: "Deploy All The Things Even If They Aren't YAML"
+image: "/assets/images/k8s-custom-deploy/title.jpg"
+description: "Integrate Tilt with custom Kubernetes deployment scripts & tools"
+tags:
+  - kubernetes
+  - tilt
+  - helm
+---
+
+If you've used Tilt, chances are you are well acquainted with the [`k8s_yaml`][api-k8s_yaml] and [`k8s_resource`][api-k8s_resource] Tiltfile functions.
+The [`k8s_yaml`][api-k8s_yaml] function deploys entities to Kubernetes from either a file or string.
+The [`k8s_resource`][api-k8s_resource] function configures port forwards and other behaviors for the deployed entities.
+
+In the Kubernetes landscape, [Helm][helm] is very popular, especially for packaging and distributing external dependencies such as nginx or Kafka.
+Tilt has a built-in [`helm`][api-helm] function for working with local charts and a [`helm_remote`][ext-helm_remote] extension for external charts stored in a Helm repo.
+
+Behind the scenes, both of these invoke `helm template` via CLI to get a locally-rendered version of the chart YAML to pass to the [`k8s_yaml`][api-k8s_yaml] built-in Tilt function.
+That is, the actual deployment to Kubernetes is done via Tilt using the Kubernetes API rather than Helm itself.
+
+More generally, this is the pattern we recommend for integrating custom Kubernetes deploy tools with Tilt, and for a lot of local development use cases, this works great!
```

Comment:
It's a bit confusing that you say "integrating custom K8s deploy tools" while the only pattern I see is to deploy helm. I'd suggest to either change to "for integrating helm deploys", or expand a bit on what pattern you mean (i.e. give more examples how to deploy other than with helm)